forge-ai/src/main/java/forge/ai/ability/AttachAi.java

```diff
@@ -78,7 +77,7 @@ protected boolean canPlayAI(Player ai, SpellAbility sa) {
             }
         }
 
-        if (!ai.getGame().getStaticEffects().getGlobalRuleChange(GlobalRuleChange.noLegendRule)
+        if (!source.ignoreLegendRule()
```

Comment:
You can remove the checks for `isLegendary()` here and in ChangeZone/PermanentAi to clean it up a bit

Reply:
can the check with `sa instanceof SpellPermanent` even work? currently Aura Spells aren't SpellPermanent